.DS_Store → macOS system file storing folder display metadata.

Thumbs.db → Windows system file storing thumbnail cache of images in a folder.

`input_dir = "spiderverse_dataset"`
**Output Directory:**  
  - `output_dir = "clean_dataset"` is where **preprocessed images** will be saved.  
  - `os.makedirs(output_dir, exist_ok=True)` ensures the folder exists or creates it.  

- **Image Size:**  
  - `IMG_SIZE = 224` sets a **uniform size** for all images (224 × 224 pixels) for machine learning models.  

In [5]:
import os
import cv2
from PIL import Image

output_dir = "clean_dataset"

IMG_SIZE = 224

os.makedirs(output_dir, exist_ok=True)

### Dataset Preprocessing Steps

- **Input Dataset:** `spiderverse_dataset` contains subfolders for each class/label.   
- **Corrupted / Hidden Files**:.DS_Store files were detected and removed automatically.
- **Folder Check:** Ensure only directories (classes) are processed.  
- **Create Output Folder:** Preserve class structure in the cleaned dataset directory.  
- **Open and Convert Images:** Convert all images to **RGB** format for consistency.  
- **Resize Images:** Resize all images to a fixed size (`IMG_SIZE × IMG_SIZE`) suitable for ML models.  
- **Save Images:** Store the cleaned images in the output folder.  
- **Handle Corrupted Images:** Skip and remove images that cannot be opened or processed.  
- **Count Images:** Keep track of successfully processed images per class.  
- **Final Output:** Cleaned, uniform, and ready-to-use dataset with preserved class structure.

In [ ]:

input_dir = "spiderverse_dataset"

for class_name in os.listdir(input_dir):


    class_path = os.path.join(input_dir, class_name)
    if not os.path.isdir(class_path):
        continue

    save_class = os.path.join(output_dir, class_name)
    os.makedirs(save_class, exist_ok=True)

    count = 0
    for img_name in os.listdir(class_path):
        img_path = os.path.join(class_path, img_name)
        try:
            img = Image.open(img_path).convert("RGB")
            img = img.resize((IMG_SIZE, IMG_SIZE))
            save_path = os.path.join(save_class, img_name)
            img.save(save_path)
            count += 1
        except Exception as e:
            print(f"Corrupted image removed: {img_path} ({e})")

    print(f"Class '{class_name}': {count} images cleaned and saved.")

Class 'spider_man_noir': 110 images cleaned and saved.
Class 'peter_b_parker': 109 images cleaned and saved.
Class 'gwen_stacy': 103 images cleaned and saved.
Corrupted image removed: spiderverse_dataset/miles_morales/.DS_Store (cannot identify image file 'spiderverse_dataset/miles_morales/.DS_Store')
Class 'miles_morales': 100 images cleaned and saved.
Class 'miguel_ohara': 110 images cleaned and saved.
Class 'spider_ham': 105 images cleaned and saved.
Class 'pavitr_prabhakar': 107 images cleaned and saved.
Corrupted image removed: spiderverse_dataset/hobie_brown/.DS_Store (cannot identify image file 'spiderverse_dataset/hobie_brown/.DS_Store')
Class 'hobie_brown': 89 images cleaned and saved.


Corrupted / Hidden Files:

.DS_Store files were detected and removed automatically.

No other corrupted images were found.

Image Preprocessing:

All images were resized to 224×224 pixels.

Converted all images to RGB format for CNN compatibility.

Class Distribution:

Classes are reasonably balanced (from 89 to 110 images per class).

Minimal imbalance remains, can be further corrected with augmentation if needed.

| Class Name       | Images Cleaned | Notes                                   |
| ---------------- | -------------- | --------------------------------------- |
| spider_man_noir  | 110            | ✅ All images cleaned                    |
| peter_b_parker   | 109            | ✅ All images cleaned                    |
| gwen_stacy       | 103            | ✅ All images cleaned                    |
| miles_morales    | 100            | ⚠️ 1 corrupted file (.DS_Store) removed |
| miguel_ohara     | 110            | ✅ All images cleaned                    |
| spider_ham       | 105            | ✅ All images cleaned                    |
| pavitr_prabhakar | 107            | ✅ All images cleaned                    |
| hobie_brown      | 89             | ⚠️ 1 corrupted file (.DS_Store) removed |
